src/trace/trace.c

```diff
@@ -486,11 +486,11 @@ void trace_on(void)
 	struct trace *trace = trace_get();
 	uint32_t flags;
 
-	spin_lock_irq(&trace->lock, flags);
-
-	trace->enable = 1;
+	/* should not do this with trace->lock held as there is trace calling internal */
```

Comment:
I was not able to parse the sentence above. 'there is trace calling internal' -> missing a complement. Or that was 'trace calling internally', which doesn't make more sense.